toot/tui/poll.py

```diff
@@ -0,0 +1,98 @@
+import urwid
+
+from toot import api
+from toot.exceptions import ApiError
+from toot.utils import format_content
+from .utils import highlight_hashtags, parse_datetime
+from .widgets import Button, CheckBox, RadioButton
+
+
+class Poll(urwid.ListBox):
+    """View and vote on a poll"""
+
+    def __init__(self, app, user, status):
+        self.status = status
+        self.app = app
+        self.user = user
+        self.poll = status.data.get("poll")
+        self.button_group = []
+
+        self.setup_listbox()
+
+    def setup_listbox(self):
+        actions = list(self.generate_contents(self.status))
+        walker = urwid.SimpleListWalker(actions)
+        super().__init__(walker)
+
+    def build_linebox(self, contents):
+        contents = urwid.Pile(list(contents))
+        contents = urwid.Padding(contents, left=1, right=1)
+        return urwid.LineBox(contents)
+
+    def vote(self, button):
+        poll = self.status.data.get("poll")
+        choices = []
+        for idx, b in enumerate(self.button_group):
```

Comment:
I prefer having more verbose variable names, unless it's a problem with line width. So `button` instead of `b` here makes it a bit more readable.